pyproject.toml

```diff
@@ -137,7 +137,7 @@ preview = true
 # This enables us to use the explicit preview rules that we want only
 explicit-preview-rules = true
 # all rules can be found here: https://docs.astral.sh/ruff/rules/
-extend-select = ["W", "I", "CPY001", "RUF"]
+extend-select = ["E", "W", "I", "CPY001", "RUF"]
```

Comment:
I think it's best to explicitly enable `E501`, not all `E` rules:
* `E1` rules are about indentation. They are redundant when using a formatter.
* `E2` rules are about missing or extra spaces. They are redundant when using a formatter.
* `E3` rules are about missing or extra lines. They are redundant when using a formatter.
* `E4` rules are enabled by default.
* `E501` can be explicitly enabled as discussed.
* [`E502`](https://docs.astral.sh/ruff/rules/redundant-backslash/) is about extra backslahes. It is redundant when using a formatter.
* `E7` rules are enabled by default.
* `E9` rules are enabled by default.

I recommend letting ruff do the right thing and enabling only `E501`.

Reply:
Yeah I wasn't sure about it but let's keep it more specific indeed for now.

One question: should we removed `"E111", "E114", "E117"` from our ignore list since there are not enabled in `ruff` by default?

Ruff default that `extend-select` extends is `["E4", "E7", "E9", "F"]`.

Reply:
I have copied over all of the [Conflicting lint rules](https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules), whether their respective ruleset is currently selected or not - just in case a ruleset is selected in the future. That's perhaps overkill, feel free to remove from teh ignore list rules from unselected rulesets.